etna/articles/models.py

```diff
@@ -84,12 +86,21 @@ class TaggedArticle(ItemBase):
     )
 
 
-class ArticlePage(HeroImageMixin, ContentWarningMixin, BasePageWithIntro):
+class ArticlePage(
+    HeroImageMixin, ContentWarningMixin, NewLabelMixin, BasePageWithIntro
+):
     """ArticlePage
 
     The ArticlePage model.
     """
 
+    def with_content_json(self, content):
+        obj = super().with_content_json(content)
+        if self.live and obj.mark_new_on_next_publish:
+            obj.mark_new_on_next_publish = False
+            obj.newly_published_at = datetime.now().date()
+        return obj
```

Comment:
I think we can simplify this slightly and always set `obj.newly_published_at` when the `obj.mark_new_on_next_publish` is `True`.

With regards to setting `obj.mark_new_on_next_publish` to `False `, if that is done here, the value wouldn't be preserved accross multiple draft version. Instead, we should do this elsewhere, when we know the 'publish' action is happening. Can you think of a good place for that?

Reply:
Also, this behaviour should be implemented on the mixin, so that we only need to maintain a single version

Reply:
> With regards to setting `obj.mark_new_on_next_publish` to `False `, if that is done here, the value wouldn't be preserved accross multiple draft version. Instead, we should do this elsewhere, when we know the 'publish' action is happening. Can you think of a good place for that?

Hi Andy, I've made the other changes that you requested but can't think where to put this? I've had a look at https://github.com/wagtail/wagtail/blob/main/wagtail/models/__init__.py#L539 but the only thing I can think of is checking if `has_unpublished_changes`, to not set it back to False - but I tried this and it works for the draft - but then it doesn't work when trying to publish.

Reply:
@jamesbiggs Moving everything to a `save()` method override would probably be the simplest solution. You should be able to look at `self.id` and `self.live` to figure out what you need.

Reply:
Hi @ababic I believe my latest changes should cover all the conversations :)